frontend/src/Shop/components/Payments.tsx

```diff
@@ -0,0 +1,47 @@
+import axios from 'axios';
+import { WindowWithEnv, PaymentDTO } from "./Types";
+
+
+const _window: WindowWithEnv = window;
+const backendURL = _window.__ENV && _window.__ENV.backendURL;
+
+const axiosClient = axios.create({ baseURL: `${backendURL}`, timeout: 20000, withCredentials: true});
+const config = {headers: {'Content-Type': 'application/json', 'Access-Control-Allow-Origin': '*'}};
+
+  
+/* 
+* logic for all of the payment callback functions that can be called
+* by the front end Pi SDK. See https://github.com/pi-apps/pi-platform-docs/blob/master/SDK_reference.md#callbacks-keys
+* for more information.
+*
+* implement your own logic as needed
+*/
+
+export const onIncompletePaymentFound = (payment: PaymentDTO) => {
+    console.log("onIncompletePaymentFound", payment);
+    return axiosClient.post('/payments/incomplete', {payment});
+  }
+  
+export const onReadyForServerApproval = (paymentId: string) => {
+    console.log("onReadyForServerApproval", paymentId);
+    axiosClient.post('/payments/approve', {paymentId}, config);
+  }
+  
+export const onReadyForServerCompletion = (paymentId: string, txid: string) => {
+    console.log("onReadyForServerCompletion", paymentId, txid);
+    axiosClient.post('/payments/complete', {paymentId, txid}, config);
+  }
+  
+export const onCancel = (paymentId: string) => {
+    console.log("onCancel", paymentId);
+    return axiosClient.post('/payments/cancelled_payment', {paymentId});
+  }
+  
+export const onError = (error: Error, payment?: PaymentDTO) => {
+    console.log("onError", error);
+    if (payment) {
+      console.log(payment);
+      // handle the error accordingly
+    }
```

Comment:
else console.error(error)

Reply:
This block is not designed to have an else statement, the if statement handles the condition where the onError callback is invoked and there was a PaymentDTO with the call. If there is no paymentDTO included then there is nothing for the callback to do

Reply:
https://github.com/pi-apps/demo/pull/95#issue-1874305511